x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/DownsampleStep.java

```diff
@@ -62,72 +72,63 @@ public void performAction(
+        IndexMetadata downsampleIndexMetadata = currentState.metadata().index(downsampleIndexName);
+        if (downsampleIndexMetadata != null) {
+            IndexMetadata.DownsampleTaskStatus downsampleIndexStatus = IndexMetadata.INDEX_DOWNSAMPLE_STATUS.get(
+                downsampleIndexMetadata.getSettings()
             );
-            // Rollup index has already been created with the generated name and its status is "success".
-            // So we skip index rollup creation.
-            if (IndexMetadata.DownsampleTaskStatus.SUCCESS.equals(rollupIndexStatus)) {
+            if (IndexMetadata.DownsampleTaskStatus.SUCCESS.equals(downsampleIndexStatus)) {
+                // Downsample index has already been created with the generated name and its status is "success".
+                // So we skip index downsample creation.
                 logger.warn(
-                    "skipping [{}] step for index [{}] as part of policy [{}] as the rollup index [{}] already exists",
+                    "skipping [{}] step for index [{}] as part of policy [{}] as the downsample index [{}] already exists",
                     DownsampleStep.NAME,
                     indexName,
                     policyName,
                     downsampleIndexName
                 );
                 listener.onResponse(null);
             } else {
-                logger.warn(
-                    "[{}] step for index [{}] as part of policy [{}] found the rollup index [{}] already exists. Deleting it.",
-                    DownsampleStep.NAME,
-                    indexName,
-                    policyName,
-                    downsampleIndexName
+                // Downsample index has already been created with the generated name but its status is not "success".
+                // So we fail this step so that we go back to cleaning up the index and try again with a new downsample
+                // index name.
+                downsampleFailed = true;
+                listener.onFailure(
+                    new IllegalStateException(
+                        "failing ["
+                            + DownsampleStep.NAME
+                            + "] step for index ["
+                            + indexName
+                            + "] as part of policy ["
+                            + policyName
+                            + "] because the downsample index ["
+                            + downsampleIndexName
+                            + "] already exists with downsample status ["
+                            + downsampleIndexStatus
+                            + "]"
+                    )
                 );
-                // Rollup index has already been created with the generated name but its status is not "success".
-                // So we delete the index and proceed with executing the rollup step.
-                DeleteIndexRequest deleteRequest = new DeleteIndexRequest(downsampleIndexName);
```

Comment:
This is removed because `cleanupDownsampleIndexKey` will be invoked in case of failure?

Reply:
Yes, the step will fail and ILM will go to the cleanup step. So no need to delete the index here